### PR TITLE
chore: Prepare rust-libp2p v0.50.0 release

### PR DESCRIPTION
**Pull request by mxinden**
## Notes

I propose we cut a new release, namely `v0.50.0`.

I included a small paragraph highlighting quic, tls and webrtc. Happy to highlight additional user-facing changes. Proposals welcome.

[Milestone 4](https://github.com/libp2p/rust-libp2p/milestone/4) tracks the outstanding items before we can cut the release. In case you would like to see something included not yet part of milestone 4, please speak up.

### CODE REVIEW

File: transports/webrtc/Cargo.toml
Comment (mxinden) [inline transports/webrtc/Cargo.toml:3]: Note that this is necessary, given that there is already `v0.3.0` under the `libp2p-webrtc` crate on crates.io. See https://github.com/libp2p/rust-libp2p/pull/3163.

//CC @melekes

**Review by thomaseizinger — APPROVED** [review]

**Comment by melekes** [comment]
> [Milestone 4](https://github.com/libp2p/rust-libp2p/milestone/4) tracks the outstanding items before we can cut the release. In case you would like to see something included not yet part of milestone 4, please speak up.

https://github.com/libp2p/rust-libp2p/issues/3157 not sure this is a blocker, but I'd love to get it fixed before 0.50

**Comment by rkuhn** [comment]
could we also include #3128 ?

**Comment by elenaf9** [comment]
I'd like to get #3144 fixed before releasing QUIC. I am currently working on it. But I doesn't have to be a blocker.

**Comment by geotro** [comment]
QUIC together with dcutr would be a big deal...

https://github.com/libp2p/rust-libp2p/issues/3143#issuecomment-1320437498_

**Comment by dignifiedquire** [comment]
any chance for #2712 to make it?

**Comment by mxinden** [comment]
I suggest we only block releases on bug fixes, e.g. https://github.com/libp2p/rust-libp2p/issues/3144. Blocking on features I made the experience that the release process drags along for a long time. Once any of the above features are merged, I am happy to cut a follow-up release, in case that is helpful for folks.

> > [Milestone 4](https://github.com/libp2p/rust-libp2p/milestone/4) tracks the outstanding items before we can cut the release. In case you would like to see something included not yet part of milestone 4, please speak up.
> 
> #3157 not sure this is a blocker, but I'd love to get it fixed before 0.50

I would not consider this a bug and thus suggest not blocking on it. In case it makes it in, great. In case it doesn't, again happy to cut a follow-up release.

> could we also include #3128 ?

I will take a look. I don't consider this a bug fix though.

> I'd like to get #3144 fixed before releasing QUIC. I am currently working on it. But I doesn't have to be a blocker.

Bug fix. Sounds good to me.

> QUIC together with dcutr would be a big deal...
> 
> [#3143 (comment)](https://github.com/libp2p/rust-libp2p/issues/3143#issuecomment-1320437498)_

@geotro this is a larger endeavor and thus will not make it in time.

> any chance for #2712 to make it?

Not a bug, thus I suggest not blocking on it. That said, in case it makes it in time, great!

//CC @kpp and @melekes as this requires adjustments on the Substrate side.

**Review by jxs — APPROVED** [review]

**Comment by mxinden** [comment]
Tagged and published the following crates:

- libp2p-autonat-v0.9.0
- libp2p-core-v0.38.0
- libp2p-dcutr-v0.8.0
- libp2p-deflate-v0.38.0
- libp2p-dns-v0.38.0
- libp2p-floodsub-v0.41.0
- libp2p-gossipsub-v0.43.0
- libp2p-identify-v0.41.0
- libp2p-kad-v0.42.0
- libp2p-mdns-v0.42.0
- libp2p-metrics-v0.11.0
- libp2p-mplex-v0.38.0
- libp2p-muxer-test-harness-v0.1.0
- libp2p-noise-v0.41.0
- libp2p-ping-v0.41.0
- libp2p-plaintext-v0.38.0
- libp2p-pnet-v0.22.2
- libp2p-quic-v0.7.0-alpha
- libp2p-relay-v0.14.0
- libp2p-rendezvous-v0.11.0
- libp2p-request-response-v0.23.0
- libp2p-swarm-derive-v0.30.2
- libp2p-swarm-v0.41.0
- libp2p-tcp-v0.38.0
- libp2p-tls-v0.1.0-alpha
- libp2p-uds-v0.37.0
- libp2p-wasm-ext-v0.38.0
- libp2p-webrtc-v0.4.0-alpha
- libp2p-websocket-v0.40.0
- libp2p-yamux-v0.42.0
- multistream-select-v0.12.1
- prost-codec-v0.3.0
- v0.50.0

**Pull request merged**
